src/leverage.ts

```diff
@@ -0,0 +1,29 @@
+import { Position, QuoteAPIResponseBody } from './types';
+import * as apisdk from '@protocolink/api';
+import { buildRouterTransactionRequest, quote } from './api';
+import * as common from '@protocolink/common';
+
+export interface LeverageParams {
+  account: string;
+  token: common.TokenObject;
+  amount: string;
+  slippage?: number;
+}
+
+export interface LeverageQuotation {
+  leverageTimes: string;
+  currentPosition: Position;
+  targetPosition: Position;
+}
+
+export async function getLeverageQuotation(
+  chainId: number,
+  marketId: string,
+  params: LeverageParams
+): Promise<QuoteAPIResponseBody<LeverageQuotation>> {
+  return quote(chainId, marketId, 'leverage', params);
+}
+
+export async function buildLeverageTransactionRequest(routerData: apisdk.RouterData) {
```

Comment:
The original idea was to allow the developers to directly call the function with the same name as the operation after getting the quotation.
Of course, if deemed unnecessary, it can also be documented that developers can directly call `buildRouterTransactionRequest` after obtaining the quotation for the operation.

Reply:
Understood.